tests/security/audit/ausearch.pm

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-Later
+#
+# Summary: Verify the "ausearch" utility can search the audit log file for certain events using various keys or
+#          other characteristics of the logged format
+# Maintainer: llzhao <llzhao@suse.com>, shawnhao <weixuan.hao@suse.com>
+# Tags: poo#81772
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $tmp_output = '/tmp/out';
+    my $tmp_backup = '/tmp/backup';
+    my $audit_log = '/var/log/audit/audit.log';
+
+    select_console 'root-console';
+
+    # Make sure audit service is started
+    assert_script_run('systemctl is-active auditd');
+
+    # Generate audit records for testing
+    assert_script_run("echo '' > $audit_log");
+    assert_script_run('systemctl stop apparmor');
+    assert_script_run('systemctl start apparmor');
+
+    # Search for an event based on the given filename
+    assert_script_run("ausearch -f /etc > $tmp_output");
+
+    # Extract pid from output log
+    script_run("tail -1 $tmp_output > $tmp_backup");
+    script_run("cat $tmp_backup | cut -d '=' -f 9 > $tmp_output");
+    my $pid = script_output("cat $tmp_output | cut -d ' ' -f 1");
+
+    # Search for an event matching the given process ID
+    assert_script_run("ausearch -p $pid > $tmp_output");
+
+    # Extract event id from output log
+    script_run("tail -1 $tmp_output > $tmp_backup");
+    script_run("awk -F: '{print \$2}' $tmp_backup > $tmp_output");
+    my $event_id = script_output("cat $tmp_output | cut -d ')' -f 1");
+
+    # Search for an event based on the given event ID
+    assert_script_run("ausearch --event $event_id");
+
+    # Search for events based on a specific CPU architecture
+    assert_script_run('ausearch --arch x86_64');
+}
```

Comment:
Could you please check the outputs/audit-records of "ausear --$options" are/is as expected?
Because the expected audit record is the most important one we wanted.
Using "validate_script_output()" you can find some example in lots of "*.pm".

Reply:
Done

Reply:
Changed to validate_script_output now. Will validate specific fields in audit log output.